lib/trino-orc/src/main/java/io/trino/orc/metadata/OrcMetadataWriter.java

```diff
@@ -128,13 +150,28 @@ public int writeFooter(SliceOutput output, Footer footer)
                         .map(OrcMetadataWriter::toUserMetadata)
                         .collect(toList()));
 
-        if (!useLegacyVersion) {
-            builder.setWriter(PRESTO_WRITER_ID);
-        }
+        setWriter(builder);
 
         return writeProtobufObject(output, builder.build());
     }
 
+    private void setWriter(OrcProto.Footer.Builder builder)
+    {
+        switch (writerIdentification) {
+            case LEGACY_HIVE_COMPATIBLE:
```

Comment:
Set writer version explicitly here for code completeness?
Otherwise add comment where it's being set.

Reply:
it would be a behavioral change, please compare this with the code behavior before my commit

Reply:
AFAICT, setting null and not setting anything is not the same thing